lib/ImportStatement.js

```diff
@@ -96,6 +102,9 @@ export default class ImportStatement {
 
     const importFunction = match.importFunction;
 
+    // Assume that an import with no assignment has side-effects.
+    const hasSideEffects = !match.assignment;
```

Comment:
I was in-between on this. There are two possible indications of an import being included for its side-effects at this point - an existing side-effect only type of import or a specified environment included it in moduleSideEffectImports. If either of those has occurred, your earlier discussions convinced me that it should be sorted to the top because it has side-effects of an unknown nature.

I then realized that there is no technical reason why an import that has a side-effect and should be sorted to the top might not also resolve an unknown variable. This left me in the boat of having an import that is both known to have side effects and exports. Part of me says that the "hasSideEffrects" should stay true even though it is not "forSideEffectsOnly" because of your  earlier arguments that it would be safest to sort it to the top. Another part says to just forget about the side effects,,, but then I think that later code could choose to resolve that unknown via other modules that export it???